src/batch/src/executor/table_function.rs

```diff
@@ -112,12 +81,120 @@ where
+
+        if let (Some(start), Some(stop), Some(step)) = (start, stop, step) {
+            Ok(GenerateSeries::<NaiveDateTimeArray, IntervalArray> { start, stop, step })
+        } else {
+            Err(ErrorCode::InternalError(
+                "the parameters of Generate Series Function are incorrect".to_string(),
+            )
+            .into())
+        }
+    }
+
+    fn new_int_generate_series(
+        array_refs: Vec<ArrayRef>,
+    ) -> Result<GenerateSeries<I32Array, I32Array>> {
+        let start = array_refs[0].clone().as_int32().value_at(0);
+        let stop = array_refs[1].clone().as_int32().value_at(0);
+        let step = array_refs[2].clone().as_int32().value_at(0);
+
+        if let (Some(start), Some(stop), Some(step)) = (start, stop, step) {
+            Ok(GenerateSeries::<I32Array, I32Array> { start, stop, step })
+        } else {
+            Err(ErrorCode::InternalError(
+                "the parameters of Generate Series Function are incorrect".to_string(),
+            )
+            .into())
+        }
+    }
+}
+
+pub struct Unnest {
+    array_refs: Vec<ArrayRef>,
+    return_type: DataType,
+}
+
+impl Unnest {
+    #[try_stream(boxed, ok = DataChunk, error = RwError)]
+    async fn eval(self) {
+        let sum = self.array_refs.iter().map(|a| a.len()).sum();
+        let mut builder = self.return_type.create_array_builder(sum)?;
+        self.array_refs
+            .iter()
+            .try_for_each(|a| builder.append_array(a))?;
+        let arr = builder.finish()?;
+        let len = arr.len();
+        let columns = vec![Column::new(Arc::new(arr))];
+        let chunk: DataChunk = DataChunk::new(columns, len);
+
+        yield chunk;
+    }
+}
+
+pub struct TableFunctionExecutor {
+    schema: Schema,
+    identity: String,
+    table_function: TableFunction,
+}
+
+impl Executor for TableFunctionExecutor {
+    fn schema(&self) -> &Schema {
+        &self.schema
+    }
+
+    fn identity(&self) -> &str {
+        &self.identity
+    }
+
+    fn execute(self: Box<Self>) -> BoxedDataChunkStream {
+        match self.table_function {
+            TableFunction::GenerateSeriesTime(gs) => gs.eval(),
+            TableFunction::GenerateSeriesInt(gs) => gs.eval(),
+            TableFunction::Unnest(us) => us.eval(),
+        }
+    }
 }
 
-pub struct GenerateSeriesExecutorBuilder {}
+pub struct TableFunctionExecutorBuilder {}
+
+impl TableFunctionExecutorBuilder {
+    pub fn new_generate_series(
+        array_refs: Vec<ArrayRef>,
+        return_type: DataType,
+    ) -> Result<(Schema, TableFunction)> {
+        match return_type {
+            DataType::Timestamp => {
+                let schema = Schema::new(vec![Field::unnamed(DataType::Timestamp)]);
+                let func =
+                    GenerateSeries::<NaiveDateTimeArray, IntervalArray>::new_time_generate_series(
+                        array_refs,
+                    )?;
```

Comment:
This is not the right way. Because the returned type has been specified in the function signature, it doesn't have to duplicate here.

```rust
fn new_time_generate_series() -> Result<TableFunction>
```